pkg/printer/options.go

```diff
@@ -98,6 +98,20 @@ func IgnoreStderr() Option {
 	}
 }
 
+// WithColor set the color mode. The when argument is one of:
+// - "auto": color mode is enabled when the standard output is connected to a
+//   terminal.
+// - "always": color mode is enabled no matter to standard output.
+// - "never": color mode is always disabled.
+// Any other value of when means "auto", which is the default.
+// The color mode is only applied with in Dict or Compact mode. For any other
+// mode, color is always disabled.
+func WithColor(when string) Option {
+	return func(_ *Options) {
+		setColors(when)
```

Comment:
Perhaaps we should store the option here, and attach the coloring to the `Printer struct` so there is no global state?

Reply:
I wasn't specifically referring to the library usage (sometimes it forces global state like here), I was more hinting that we shouldn't add our own global `func setColors`. We shouldn't expose the innards of the library in our own code, as that gives us some more flexibility to even swap to a different library in the future without much impact.

I was thinking we can even add our own custom `struct color(er?)` and attach to options/printer to completely hide which library is even used.

---

Just out of curiosity, why is `DisableColor()` on each color not an option? All we'd really have to do is wrap all the colors into a some sort of a collection and iterate over them, no?

Reply:
Valid point. I'll rework the PR, see what I can come up with.

Reply:
I reworked the PR to avoid using globals. The new `colorer` struct could easily be adapted for a new library if need be. PTAL.